### PR TITLE
display: Add support for `AIP31068` based displays

### PR DESCRIPTION
**Pull request by bazarovdev**
Adds support for `YHCB2004` display used in `Geeetech A10/M/T` printers. 
The display is based on `AIP31068` controller and uses 9-bit SPI commands. 
The details were found in Marlin FW that supports this display (see https://github.com/MarlinFirmware/Marlin/issues/25196).

~~This PR depends on merging of https://github.com/Klipper3d/klipper/pull/6633 to enable 9-bit SPI.~~
UPD: implemented proposed fix of sending 8 commands (9 bit each) = 9 bytes (8 bit each)

### CODE REVIEW

**Comment by bazarovdev** [comment]
Added padding to 8 commands and sending them as 9 byte sequence (as proposed in PR below)
It eliminates need to change SW SPI driver (https://github.com/Klipper3d/klipper/pull/6633)

Tested on YHCB2004 display of Geeetech A10M:
image

**Comment by bazarovdev** [comment]
Self review:

**Free of defects:** adds new dispaly driver. Attached image from testing on YHCB2004 display.
**Impact:** gives option to integrate support to Geeetech A10/M/T, ender clones with mixing extruders (M=2in1out, T=3in1out) increasing user base of Klipper and potentially helping test features related to mixing.
(Currently, most users stuck with Marlin or mainly replace stock board+display to work with Klipper. The facebook group of owners contains 6.3k users).
**Licensing:** 1 file introduced with GNU GPLv3 licensing, commit signed-off.
**Guidelines:** Tried to follow to my best understanding.
**Docs:** PR contains update to docs related usage of new driver (same as `hd44780_spi`)
**Commits:** 1 commit with 1 feature.

File: docs/Config_Reference.md [hunk 1]
```diff
@@ -4215,6 +4216,32 @@ spi_software_miso_pin:
 ...
 ```
 
+#### aip31068_spi display
+
+Information on configuring an aip31068_spi display - a very similar to
+hd44780_spi display but uses 9 bit SPI commands.
+
+To avoid modifying SW_SPI driver instead sending 9-bit command, 8 commands packed in series and it sums to full 9 bytes without need for exoctic drivers.
```
Comment (KevinOConnor) [inline docs/Config_Reference.md:4224]: FWIW, I don't think the config reference is a good place to describe the implementation details.  Users configuring the display wont care how it is implemented.

File: klippy/extras/display/aip31068_spi.py [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+# Copyright (C) 2024  Alexander Bazarov <oss@bazarov.dev>
+#
+# based on : hd44780_spi.py
+# Copyright (C) 2018  Kevin O'Connor <kevin@koconnor.net>
+# Copyright (C) 2018  Eric Callahan <arksine.code@gmail.com>
+# Copyright (C) 2021  Marc-Andre Denis <marcadenis@msn.com>
+#
+# inspired by https://github.com/red-scorp/LiquidCrystal_AIP31068
+#
+# This file may be distributed under the terms of the GNU GPLv3 license.
```
Comment (KevinOConnor) [inline klippy/extras/display/aip31068_spi.py:12]: Please format the top of the file with 1-line file overview comment, 1-line comment blank, n-line copyrights, 1-line comment blank, 1-line license statement.  It's confusing to list "inspired" by or "based on" comments in the copyright block.  Either some entity has claimed a copyright or not.

Don't feel the need to put my copyright on this file.

Feel free to add contextual information in comments outside of the main copyright block.

**Review by KevinOConnor — COMMENTED** [review]
Thanks.  In general it seems fine to me.  I have a couple of minor comments on the documentation.

Cheers,
-Kevin

File: docs/Config_Reference.md (hunk 1 above)
Comment (bazarovdev) [reply docs/Config_Reference.md]: removed

**Comment by bazarovdev** [comment]
@KevinOConnor 
Thanks for review, commited fixes, but feel free to edit if it is still not in needed style.

If merged, will prepare printer config for submission.
Thanks,
Alex

**Comment by KevinOConnor** [comment]
Okay thanks.  I was able to merge this.  Unfortunately, github did something weird and the commit message got lost.  Sorry about that.

-Kevin

**Comment by bazarovdev** [comment]
Thanks, I don't see what is got lost, so never mind.

my first PR to Klipper got merged :tada: 
1 year from idea->fixed on christmas holidays->rewritten to avoid sw_spi changes->approved->merged. 
A metric for me to work on :chart_with_upwards_trend: :)

**Pull request merged**